class-two-factor-core.php

```diff
@@ -537,8 +537,8 @@ public static function show_two_factor_login( $user ) {
 	 */
 	public static function backup_2fa() {
 		$wp_auth_id = filter_input( INPUT_GET, 'wp-auth-id', FILTER_SANITIZE_NUMBER_INT );
-		$nonce      = filter_input( INPUT_GET, 'wp-auth-nonce', FILTER_SANITIZE_STRING );
-		$provider   = filter_input( INPUT_GET, 'provider', FILTER_SANITIZE_STRING );
+		$nonce      = filter_input( INPUT_GET, 'wp-auth-nonce', FILTER_CALLBACK, [ 'options' => 'sanitize_key' ] );
+		$provider   = filter_input( INPUT_GET, 'provider', FILTER_CALLBACK, [ 'options' => 'sanitize_text_field' ] );
```

Comment:
`$provider` can contain uppercase letters; therefore, `sanitize_key()` is not suitable here.